test/teos/unit/test_chain_monitor.py

```diff
@@ -13,148 +15,245 @@ def test_init(block_processor):
     # run_bitcoind is started here instead of later on to avoid race conditions while it initializes
 
     # Not much to test here, just sanity checks to make sure nothing goes south in the future
-    chain_monitor = ChainMonitor(Queue(), Queue(), block_processor, bitcoind_feed_params)
+    chain_monitor = ChainMonitor([Queue(), Queue()], block_processor, bitcoind_feed_params)
 
+    assert chain_monitor.status == ChainMonitorStatus.IDLE
     assert chain_monitor.best_tip is None
     assert isinstance(chain_monitor.last_tips, list) and len(chain_monitor.last_tips) == 0
-    assert chain_monitor.terminate is False
+    assert chain_monitor.status == ChainMonitorStatus.IDLE
     assert isinstance(chain_monitor.check_tip, Event)
     assert isinstance(chain_monitor.lock, Condition)
     assert isinstance(chain_monitor.zmqSubSocket, zmq.Socket)
 
     # The Queues and asleep flags are initialized when attaching the corresponding subscriber
-    assert isinstance(chain_monitor.watcher_queue, Queue)
-    assert isinstance(chain_monitor.responder_queue, Queue)
+    assert isinstance(chain_monitor.receiving_queues[0], Queue)
+    assert isinstance(chain_monitor.receiving_queues[1], Queue)
 
 
-def test_notify_subscribers(block_processor):
-    chain_monitor = ChainMonitor(Queue(), Queue(), block_processor, bitcoind_feed_params)
-    # Subscribers are only notified as long as they are awake
-    new_block = get_random_value_hex(32)
+def test_notify_listeners(block_processor):
+    queue1 = Queue()
+    queue2 = Queue()
+    chain_monitor = ChainMonitor([queue1, queue2], block_processor, bitcoind_feed_params)
 
     # Queues should be empty to start with
-    assert chain_monitor.watcher_queue.empty()
-    assert chain_monitor.responder_queue.empty()
+    assert queue1.qsize() == 0
+    assert queue2.qsize() == 0
+
+    block1 = get_random_value_hex(32)
+    block2 = get_random_value_hex(32)
+    block3 = get_random_value_hex(32)
+
+    # we add two elements to the internal queue before the thread is started
+    chain_monitor.queue.put(block1)
+    chain_monitor.queue.put(block2)
+
+    notifying_thread = Thread(target=chain_monitor.notify_listeners, daemon=True)
+    notifying_thread.start()
+
+    # the existing elements should be processed soon and in order for all queues
+    for q in [queue1, queue2]:
+        assert q.get(timeout=0.1) == block1
+        assert q.get(timeout=0.1) == block2
 
-    chain_monitor.notify_subscribers(new_block)
+    # Subscribers are only notified as long as they are awake
```

Comment:
Since you're checking this, it may be worth checking that the queues are empty before starting the thread.

Reply:
Added in cc60fa2.